docs/index.md

```diff
@@ -1179,6 +1179,47 @@ app/components
 </div>
 ```
 
+##### Translations
+
+In order to support `I18n` translations scoped to the component you can include `ViewComponent::SidecarI18n`
+into your component:
+
+`app/components/example_component.rb`
+
+```ruby
+module ExampleComponent < ApplicationComponent
+  include ViewComponent::SidecarI18n
+end
+```
+
+This allows for adding a "sidecar" YAML file that will define translations for the component:
+
+`app/components/example_component.yml`
+
+```yml
+en:
+  # This key will be expanded to `en.example_component.hello`
+  hello: "Hello world!"
+```
+
+The translations will then be available as usual:
+
+`app/components/example_component.html.erb`
+
+```erb
+<p><%= t("hello") %></p>
+```
+
+It's important to notice that if the component needs to access global translations
+that can be done explicitly mentioning `helper` or `I18n`:
```

Comment:
```suggestion
To access global translations, explicitly mention `helper` or `I18n`:
```

That being said, I don't know if this is ideal. I feel like we shouldn't have to require folks to use two different APIs here.